packages/story-editor/src/components/floatingMenu/karma/floatingMenu.karma.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { waitFor } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../../app/story';
+import { Fixture } from '../../../karma';
+import { focusFloatingMenu, tabToCanvasFocusContainer } from './utils';
+
+describe('Design Menu: Keyboard Navigation', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+    fixture.setFlags({ floatingMenu: true });
+    await fixture.render();
+    await fixture.collapseHelpCenter();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  it('should never pass focus to floating menu when using keyboard from outside of canvas', async () => {
+    // add a shape to the canvas so that a floating menu is visible
+    await fixture.events.click(fixture.editor.library.shapesTab);
+    await waitFor(() => fixture.editor.library.shapes);
+
+    await fixture.events.click(
+      fixture.editor.library.shapes.shape('Rectangle')
+    );
+
+    // Now let's focus the footer and tab a few times. We should never run into the floating menu or its focusable content.
+
+    const { checklistToggle } = fixture.editor.footer;
+
+    await fixture.events.click(checklistToggle);
+    await fixture.events.sleep(300); // allow transition to play out
+    // close the checklist again, we're using it as a focus anchor
+    await fixture.events.click(checklistToggle);
+    // Floating menus are after the footer popups in the tab order, so if it was going to be selected it would happen after the footer content.
+    let count = 0;
+    while (count < 15) {
+      // eslint-disable-next-line no-await-in-loop -- need to await key press
+      await fixture.events.keyboard.press('tab');
+
+      expect(fixture.editor.canvas.designMenu).not.toContain(
+        document.activeElement
+      );
+
+      count++;
+    }
+  });
+
+  it('should return focus to element on esc', async () => {
```

Comment:
maybe `should return focus to element & maintain selection on esc` ? thoughts? only mentioning cus focus & selection have been causing some confusion that they can get out of sync

This tests already checks this and is awesome! just pondering a slight naming update.

Reply:
great clarification